src/codegen/codegen-expression.lisp

```diff
@@ -22,204 +25,214 @@
 
 (in-package #:coalton-impl/codegen/codegen-expression)
 
-(defgeneric codegen-expression (node current-function env)
-  (:method ((node node-literal) current-function env)
-    (declare (type tc:environment env)
-             (type (or null symbol) current-function)
-             (ignore current-function env))
-    (node-literal-value node))
-
-  (:method ((node node-variable) current-function env)
-    (declare (type tc:environment env)
-             (type (or null symbol) current-function)
-             (ignore current-function env))
-    (node-variable-value node))
-
-  (:method ((node node-application) current-function env)
-    (declare (type tc:environment env)
-             (type (or null symbol) current-function))
-    (let* ((arity (length (node-application-rands node)))
-
-           (function-applicator
-             (gethash arity *function-application-functions*)))
-
-      `(,function-applicator
-        ,(codegen-expression (node-application-rator node) current-function env)
-        ,@(mapcar
-           (lambda (node)
-             (codegen-expression node current-function env))
-           (node-application-rands node)))))
-
-  (:method ((node node-direct-application) current-function env)
-    (declare (type tc:environment env)
-             (type (or null symbol) current-function))
-    `(,(node-direct-application-rator node)
+(defgeneric codegen-expression (node current-function env))
```

Comment:
Leave this as a single `defgeneric`. That way it reads like a match expression.

Reply:
I can aesthetically appreciate wanting it to look like a match expression, but I don't think it's worth the downgrade in developer experience caused by breaking `grep defmethod foo` and `M-.` to jump to method definitions.

Reply:
That said, I don't think I actually ended up needing to edit any of the methods on either of these functions, so these changes don't belong in this PR, and I'm not going to bother making another PR just for this sort of change.

Reply:
Jump to definition still works for me with `defgeneric`
<img width="815" alt="Screen Shot 2022-06-13 at 2 26 59 PM" src="https://user-images.githubusercontent.com/6886632/173448387-0d1a7d01-5d47-4606-ba78-0958580d417b.png">
.

Reply:
:shrug: I don't know what to tell you. With SLY on SBCL, the `:method` version always jumps to the top of the `defgeneric` form. But not at issue; I'll strip these changes out of this PR as irrelevant.